sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingClientBuilder.java

```diff
@@ -0,0 +1,345 @@
+        if (this.httpClient != null && client == null) {
+            logger.info("HttpClient is being set to 'null' when it was previously configured.");
+        }
+
+        this.httpClient = client;
+        return this;
+    }
+
+    /**
+     * Sets the HTTP pipeline to use for the service client.
+     * <p>
+     * If {@code pipeline} is set, all other settings are ignored, aside from
+     * {@link FormTrainingClientBuilder#endpoint(String) endpoint} to build {@link FormTrainingAsyncClient} or
+     * {@link FormTrainingClient}.
+     *
+     * @param httpPipeline The HTTP pipeline to use for sending service requests and receiving responses.
+     *
+     * @return The updated FormRecognizerClientBuilder object.
+     */
+    public FormTrainingClientBuilder pipeline(HttpPipeline httpPipeline) {
+        if (this.httpPipeline != null && httpPipeline == null) {
+            logger.info("HttpPipeline is being set to 'null' when it was previously configured.");
+        }
+
+        this.httpPipeline = httpPipeline;
+        return this;
+    }
+
+    /**
+     * Sets the configuration store that is used during construction of the service client.
+     * <p>
+     * The default configuration store is a clone of the {@link Configuration#getGlobalConfiguration() global
+     * configuration store}, use {@link Configuration#NONE} to bypass using configuration settings during construction.
+     *
+     * @param configuration The configuration store used to
+     *
+     * @return The updated FormRecognizerClientBuilder object.
+     */
+    public FormTrainingClientBuilder configuration(Configuration configuration) {
+        this.configuration = configuration;
+        return this;
+    }
+
+    /**
+     * Sets the {@link RetryPolicy#RetryPolicy()} that is used when each request is sent.
+     * <p>
+     * The default retry policy will be used if not provided {@link FormTrainingClientBuilder#buildAsyncClient()}
+     * to build {@link FormTrainingAsyncClient} or {@link FormTrainingClient}.
+     *
+     * @param retryPolicy user's retry policy applied to each request.
+     *
+     * @return The updated FormRecognizerClientBuilder object.
+     */
+    public FormTrainingClientBuilder retryPolicy(RetryPolicy retryPolicy) {
+        this.retryPolicy = retryPolicy;
+        return this;
+    }
+
+    /**
+     * Sets the {@link FormRecognizerServiceVersion} that is used when making API requests.
+     * <p>
+     * If a service version is not provided, the service version that will be used will be the latest known service
+     * version based on the version of the client library being used. If no service version is specified, updating to a
+     * newer version the client library will have the result of potentially moving to a newer service version.
+     *
+     * @param version {@link FormRecognizerServiceVersion} of the service to be used when making requests.
+     *
+     * @return The updated FormRecognizerClientBuilder object.
```

Comment:
Fix this in all the methods of the builder.
```suggestion
     * @return The updated FormTrainingClientBuilder object.
```